### ISSUE TITLE
Taupage-init cannot start with C5/M5 instances and external volumes

### ISSUE THREAD

**Issue by nolar** [zalando-stups/taupage#492]
When trying to deploy a stack with C5/M5 type instances AND with an external volume attached, `taupage-init` fails to prepare the disks, so the app fails to start.

This is caused by C5/M5 instances detecting the attached volumes as `/dev/nvme1n1` devices, while the AWS console shows the requested `/dev/xvdk`, and `taupage-init` waits for the `/dev/xvdk` device to appear.

A sample Senza file (minimised fragment):

```yaml
SenzaComponents:
  - Configuration:
      Type: Senza::StupsAutoConfiguration
  - AppServer:
      Type: Senza::TaupageAutoScalingGroup
      InstanceType: c5.large # t2.micro
      TaupageConfig:
         runtime: Docker
         source: "................................"
         mounts:
           /data:
             partition: /dev/xvdk
             erase_on_boot: true
             filesystem: ext4
      BlockDeviceMappings:
        - DeviceName: /dev/xvdk
          Ebs:
            VolumeSize: 20
```

In the logs, it looks like:

```
/var/log/syslog Mar 28 12:06:41 ip-172-31-143-95 taupage-init: message repeated 5 times: [ INFO: Waiting for /dev/xvdk to stabilize..]
/var/log/syslog Mar 28 12:06:41 ip-172-31-143-95 taupage-init: ERROR: Failed to wait for /dev/xvdk device to become available after 60 seconds
/var/log/syslog Mar 28 12:06:41 ip-172-31-143-95 taupage-init: ERROR: ./init.d/10-prepare-disks.py did not successfully finish
```

The proper disk names can be detected by `sudo fdisk -l` or any other means. It will be `/dev/nvme1n1` for the attached volume, `/dev/nvme0n1` for the system disk, etc.

This behaviour is documented for AWS C5/M5 instances: 

* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html
* https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html

This does not happen in C4 instance types. However, C5/M5 are 10-15% more efficient and 10-15% less expensive (according to some comparisons), so they are desired.

**Comment by hjacobs** [comment]
Thanks for reporting.

**Comment by aermakov-zalando** [comment]
I'm not sure why this is a bug. Device names will be different on C5/M5 instances, but they're not hardcoded anywhere in Taupage or Senza, so the users can just update their stacks to use correct device names.

**Comment by hjacobs** [comment]
@aermakov-zalando ok, maybe we should just clearly document this?

**Comment by aermakov-zalando** [comment]
I've investigated a bit more after another user request, and it looks like we can actually fix it. The problem is that the NVMe-based instances completely ignore BlockDeviceMappings; however we can use `nvme id-ctrl` to rename them manually. Unfortunately, it's not available at all in 14.04, but we could install the 18.04 version.

**Comment by aermakov-zalando** [comment]
Returning the `bug` label, sorry for misunderstanding originally.

**Comment by Jan-M** [comment]
For newer Taupage AMIs this does work. However I am told there is a issue when attaching more than one EBS device, as the device assignment then is non deterministic.

```
        volumes:
          ebs:
            /dev/sdk: "spilo_yourdatabase_data"
        mounts:
          /home/postgres/pgdata:
            partition: /dev/nvme1n1
            filesystem: ext4
            options: noatime,nodiratime,nobarrier
            erase_on_boot: false
```

**Comment by nolar** [comment]
FYI: We do not use STUPS anymore in our team for long time, so this issue is not needed for us. Maybe for other teams — it is for you to decide on the future of this issue.

**Comment by dehora** [comment]
@mmichalski @ophelan if you had a near incident related to this can you add the details here please?

**Comment by aermakov-zalando** [comment]
Fixed by #573/#574.
